src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs

```diff
@@ -335,6 +335,11 @@ internal static void EmitKeyedServiceParameterPreparation(this EndpointParameter
     {
         codeWriter.WriteLine(endpointParameter.EmitParameterDiagnosticComment());
 
+        codeWriter.WriteLine("if (httpContext.RequestServices.GetService<IServiceProviderIsService>() is not IServiceProviderIsKeyedService)");
+        codeWriter.StartBlock();
+        codeWriter.WriteLine(@"throw new InvalidOperationException(""This service provider doesn't support keyed services."");");
```

Comment:
```suggestion
        codeWriter.WriteLine(@"throw new InvalidOperationException($""Unable to resolve {nameof(FromKeyedServicesAttribute)} because the service provider doesn't support keyed services."");");
```

Reply:
The exception message follows the pattern used in the runtime and a few other places ([ref](https://grep.app/search?q=This%20service%20provider%20doesn%27t%20support%20keyed%20services.)). It's admittedly a bit vague for the specific case but I figured it's better to deliver the same exception message across all implementations.

Reply:
That error message makes sense if you're calling `GetKeyedService` directly because the stack trace points you to the cause. In infrastructure code like this it would be better to give an indication to why a keyed service is being resolved (the attribute).

I agree on consistency. HubMethodDescriptor.cs should be improved as well.